test/integration/component/test_netscaler_lb_algo.py

```diff
@@ -76,8 +78,12 @@ def setUpClass(cls):
                 cls.testdata["service_offering"]
             )
         except Exception as e:
-            cls.tearDownClass()
-            raise Exception("Warning: Exception in setUpClass: %s" % e)
+            if cls.exception_string.lower() in e.lower():
```

Comment:
e.lower() won't work as e is an Exception object. Convert it to string and then use it.